src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +346,30 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            return HttpTelemetry.IsEnabled && request.RequestUri != null ?
+                WithLogging(handler, request, cancellationToken) :
+                handler.SendAsync(request, cancellationToken);
+
+            static async Task<HttpResponseMessage> WithLogging(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Debug.Assert(request.RequestUri != null);
+                HttpTelemetry.Log.RequestStart(
+                    request.RequestUri.IdnHost,
+                    request.RequestUri.Port,
+                    request.RequestUri.Scheme,
+                    request.RequestUri.AbsolutePath,
+                    request.Version.ToString());
+                try
+                {
+                    return await handler.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                }
+                catch
+                {
+                    HttpTelemetry.Log.RequestAbort();
+                    HttpTelemetry.Log.RequestStop();
```

Comment:
Any risk of this logging stop more than once due to the RequestStop() call in the dispose path? (I didn't scrutinize the http code, I trust you guys are in the best position to decide if this is relevant)

Reply:
Potentially, see https://github.com/dotnet/runtime/pull/37619#discussion_r441164516.

I'll work on fixing this potential issue.